<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="tidymass">
<title>Pathway enrichment â€¢ tidymass</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Pathway enrichment">
<meta property="og:description" content="tidymass">
<meta property="og:image" content="https://tidymass.github.io/tidymass/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-000000-01"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-000000-01');
</script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">tidymass</a>

    <small class="nav-text text-danger me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.99.6</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/packages.html">
    <span class="fa fa-code"></span>
     
    Packages
  </a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown--installation">
    <span class="fa fa-download"></span>
     
    Installation
  </a>
  <div class="dropdown-menu" aria-labelledby="dropdown--installation">
    <a class="dropdown-item" href="../articles/tidymass_install.html">Install tidymass</a>
    <a class="dropdown-item" href="../articles/tidymass_update.html">Update tidymass</a>
    <a class="dropdown-item" href="../articles/docker.html">Docker version of tidymass</a>
    <a class="dropdown-item" href="../articles/build_docker.html">Build your own images based on tidymass</a>
  </div>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown--tutorial">
    <span class="fa fa-folder-open"></span>
     
    Tutorial
  </a>
  <div class="dropdown-menu" aria-labelledby="dropdown--tutorial">
    <h6 class="dropdown-header" data-toc-skip>Demo data</h6>
    <a class="dropdown-item" href="../articles/demo_data.html">Download demo data</a>
    <a class="dropdown-item" href="../articles/create_mass_dataset.html">Create mass_dataset class from other tools</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Use tidymass step by step</h6>
    <a class="dropdown-item" href="../articles/data_convert.html">Convert raw data to mzXML/mgf</a>
    <a class="dropdown-item" href="../articles/raw_data_processing.html">Raw data processing</a>
    <a class="dropdown-item" href="../articles/explore_data.html">Explore data</a>
    <a class="dropdown-item" href="../articles/data_cleaning.html">Data cleaning</a>
    <a class="dropdown-item" href="../articles/metabolite_annotation.html">Metabolite annotation</a>
    <a class="dropdown-item" href="../articles/statistical_analysis.html">Statistical analysis</a>
    <a class="dropdown-item" href="../articles/pathway_enrichment.html">Pathway enrichment</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/whole_workflow.html">Use tidymass for whole workflow</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Case study for tidyMass paper</h6>
    <a class="dropdown-item" href="../articles/case_study_data.html">Download data</a>
    <a class="dropdown-item" href="../articles/tidymass_case_study.html">Case study</a>
    <a class="dropdown-item" href="../articles/repeat_case_study.html">Repeat case study by docker</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown--news-amp-blog">
    <span class="fa fa-rss"></span>
     
    News &amp; Blog
  </a>
  <div class="dropdown-menu" aria-labelledby="dropdown--news-amp-blog">
    <h6 class="dropdown-header" data-toc-skip>Blog posts</h6>
    <a class="dropdown-item" href="../articles/version0.4.0.html">Version 0.4.0</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../news/index.html">Change log</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/faq.html">
    <span class="fa fa-question-circle"></span>
     
    FAQ
  </a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown--more">
    <span class="fa fa-laptop-code"></span>
     
    More
  </a>
  <div class="dropdown-menu" aria-labelledby="dropdown--more">
    <a class="external-link dropdown-item" href="https://github.com/yufree/xcmsrocker">xcmsrocker</a>
    <a class="external-link dropdown-item" href="https://innovations.stanford.edu/sdo">Stanford Data Ocean</a>
    <a class="external-link dropdown-item" href="https://jaspershen.github.io/MetNormalizer/">MetNormalizer</a>
    <a class="external-link dropdown-item" href="https://jaspershen.github.io/lipidflow/">lipidflow</a>
    <a class="external-link dropdown-item" href="https://jaspershen.github.io/laggedcor/">laggedcor</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/tidymass/tidymass">
    <span class="fa fa-github"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://twitter.com/xiaotaoshen1990">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Pathway enrichment</h1>
                        <h4 data-toc-skip class="author">Xiaotao Shen (<a href="https://www.shenxt.info/" class="external-link uri">https://www.shenxt.info/</a>)</h4>
                        
            <h4 data-toc-skip class="date">Created on 2021-12-04 and updated on 2022-03-21</h4>
      
      
      <div class="d-none name"><code>pathway_enrichment.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="data-preparation">
<strong>Data preparation</strong><a class="anchor" aria-label="anchor" href="#data-preparation"></a>
</h2>
<p>Load the differential expressional metabolites.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">tidymass</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"statistical_analysis/object_final"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">object_final</span>
<span class="co">#&gt; <span style="color: #BBBB00;">-------------------- </span></span>
<span class="co"><span style="color: #BBBB00;">#&gt; </span><span style="color: #00BB00;">massdataset version: 0.99.14 </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span><span style="color: #BBBB00;">-------------------- </span></span>
<span class="co"><span style="color: #BBBB00;">#&gt; </span><span style="color: #00BB00;">1.expression_data:</span>[ 291 x 259 data.frame]</span>
<span class="co">#&gt; <span style="color: #00BB00;">2.sample_info:</span>[ 259 x 11 data.frame]</span>
<span class="co">#&gt; <span style="color: #00BB00;">3.variable_info:</span>[ 291 x 12 data.frame]</span>
<span class="co">#&gt; <span style="color: #00BB00;">4.sample_info_note:</span>[ 11 x 2 data.frame]</span>
<span class="co">#&gt; <span style="color: #00BB00;">5.variable_info_note:</span>[ 9 x 2 data.frame]</span>
<span class="co">#&gt; <span style="color: #00BB00;">6.ms2_data:</span>[ 2084 variables x 1902 MS2 spectra]</span>
<span class="co">#&gt; <span style="color: #BBBB00;">-------------------- </span></span>
<span class="co"><span style="color: #BBBB00;">#&gt; </span><span style="color: #00BB00;">Processing information (extract_process_info())</span></span>
<span class="co"><span style="color: #00BB00;">#&gt; create_mass_dataset ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>      Package         Function.used                Time</span>
<span class="co">#&gt; 1 massdataset create_mass_dataset() 2022-01-16 16:19:04</span>
<span class="co">#&gt; <span style="color: #00BB00;">process_data ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>        Package Function.used                Time</span>
<span class="co">#&gt; 1 massprocesser  process_data 2022-01-16 16:18:43</span>
<span class="co">#&gt; <span style="color: #00BB00;">mutate ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>      Package Function.used                Time</span>
<span class="co">#&gt; 1 massdataset      mutate() 2022-01-16 23:48:08</span>
<span class="co">#&gt; <span style="color: #00BB00;">mutate_variable_na_freq ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>      Package             Function.used                Time</span>
<span class="co">#&gt; 1 massdataset mutate_variable_na_freq() 2022-01-18 09:11:43</span>
<span class="co">#&gt; 2 massdataset mutate_variable_na_freq() 2022-01-18 09:11:43</span>
<span class="co">#&gt; 3 massdataset mutate_variable_na_freq() 2022-01-18 09:11:43</span>
<span class="co">#&gt; <span style="color: #00BB00;">filter ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>      Package Function.used                Time</span>
<span class="co">#&gt; 1 massdataset      filter() 2022-01-18 09:11:44</span>
<span class="co">#&gt; 2 massdataset      filter() 2022-03-07 17:05:56</span>
<span class="co">#&gt; 3 massdataset      filter() 2022-03-07 17:05:56</span>
<span class="co">#&gt; 4 massdataset      filter() 2022-03-07 17:05:58</span>
<span class="co">#&gt; 5 massdataset      filter() 2022-03-07 17:05:58</span>
<span class="co">#&gt; 6 massdataset      filter() 2022-03-07 17:05:58</span>
<span class="co">#&gt; 7 massdataset      filter() 2022-03-07 17:05:58</span>
<span class="co">#&gt; <span style="color: #00BB00;">impute_mv ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>      Package Function.used                Time</span>
<span class="co">#&gt; 1 masscleaner   impute_mv() 2022-01-18 09:38:02</span>
<span class="co">#&gt; <span style="color: #00BB00;">normalize_data ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>      Package    Function.used                Time</span>
<span class="co">#&gt; 1 masscleaner normalize_data() 2022-01-18 09:38:07</span>
<span class="co">#&gt; <span style="color: #00BB00;">integrate_data ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>      Package    Function.used                Time</span>
<span class="co">#&gt; 1 masscleaner integrate_data() 2022-01-18 09:38:08</span>
<span class="co">#&gt; <span style="color: #00BB00;">update_mass_dataset ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>      Package         Function.used                Time</span>
<span class="co">#&gt; 1 massdataset update_mass_dataset() 2022-01-19 21:53:01</span>
<span class="co">#&gt; <span style="color: #00BB00;">mutate_ms2 ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>      Package Function.used                Time</span>
<span class="co">#&gt; 1 massdataset  mutate_ms2() 2022-01-19 21:53:36</span>
<span class="co">#&gt; <span style="color: #00BB00;">annotate_metabolites_mass_dataset ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>  Package                       Function.used                Time</span>
<span class="co">#&gt; 1   metid annotate_metabolites_mass_dataset() 2022-02-22 21:16:23</span>
<span class="co">#&gt; 2   metid annotate_metabolites_mass_dataset() 2022-02-22 21:30:10</span>
<span class="co">#&gt; 3   metid annotate_metabolites_mass_dataset() 2022-02-22 21:47:59</span>
<span class="co">#&gt; <span style="color: #00BB00;">create_mass_dataset ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>      Package         Function.used                Time</span>
<span class="co">#&gt; 1 massdataset create_mass_dataset() 2022-01-16 16:20:02</span>
<span class="co">#&gt; <span style="color: #00BB00;">process_data ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>        Package Function.used                Time</span>
<span class="co">#&gt; 1 massprocesser  process_data 2022-01-16 16:19:48</span>
<span class="co">#&gt; <span style="color: #00BB00;">mutate ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>      Package Function.used                Time</span>
<span class="co">#&gt; 1 massdataset      mutate() 2022-01-16 23:48:08</span>
<span class="co">#&gt; <span style="color: #00BB00;">mutate_variable_na_freq ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>      Package             Function.used                Time</span>
<span class="co">#&gt; 1 massdataset mutate_variable_na_freq() 2022-01-18 09:11:47</span>
<span class="co">#&gt; 2 massdataset mutate_variable_na_freq() 2022-01-18 09:11:47</span>
<span class="co">#&gt; 3 massdataset mutate_variable_na_freq() 2022-01-18 09:11:47</span>
<span class="co">#&gt; <span style="color: #00BB00;">filter ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>      Package Function.used                Time</span>
<span class="co">#&gt; 1 massdataset      filter() 2022-01-18 09:11:47</span>
<span class="co">#&gt; 2 massdataset      filter() 2022-03-07 17:05:56</span>
<span class="co">#&gt; 3 massdataset      filter() 2022-03-07 17:05:56</span>
<span class="co">#&gt; <span style="color: #00BB00;">impute_mv ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>      Package Function.used                Time</span>
<span class="co">#&gt; 1 masscleaner   impute_mv() 2022-01-18 09:38:06</span>
<span class="co">#&gt; <span style="color: #00BB00;">normalize_data ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>      Package    Function.used                Time</span>
<span class="co">#&gt; 1 masscleaner normalize_data() 2022-01-18 09:50:47</span>
<span class="co">#&gt; <span style="color: #00BB00;">integrate_data ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>      Package    Function.used                Time</span>
<span class="co">#&gt; 1 masscleaner integrate_data() 2022-01-18 09:50:47</span>
<span class="co">#&gt; <span style="color: #00BB00;">update_mass_dataset ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>      Package         Function.used                Time</span>
<span class="co">#&gt; 1 massdataset update_mass_dataset() 2022-01-19 21:53:37</span>
<span class="co">#&gt; <span style="color: #00BB00;">mutate_ms2 ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>      Package Function.used                Time</span>
<span class="co">#&gt; 1 massdataset  mutate_ms2() 2022-01-19 21:54:06</span>
<span class="co">#&gt; <span style="color: #00BB00;">annotate_metabolites_mass_dataset ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>  Package                       Function.used                Time</span>
<span class="co">#&gt; 1   metid annotate_metabolites_mass_dataset() 2022-02-22 21:50:29</span>
<span class="co">#&gt; 2   metid annotate_metabolites_mass_dataset() 2022-02-22 22:08:10</span>
<span class="co">#&gt; 3   metid annotate_metabolites_mass_dataset() 2022-02-22 22:26:33</span>
<span class="co">#&gt; <span style="color: #00BB00;">merge_mass_dataset ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>      Package      Function.used                Time</span>
<span class="co">#&gt; 1 massdataset merge_mass_dataset 2022-03-07 17:05:56</span>
<span class="co">#&gt; <span style="color: #00BB00;">slice_head ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>      Package Function.used                Time</span>
<span class="co">#&gt; 1 massdataset  slice_head() 2022-03-07 17:05:58</span>
<span class="co">#&gt; 2 massdataset  slice_head() 2022-03-07 17:05:58</span>
<span class="co">#&gt; <span style="color: #00BB00;">mutate_fc ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>      Package Function.used                Time</span>
<span class="co">#&gt; 1 massdataset   mutate_fc() 2022-03-07 17:05:59</span>
<span class="co">#&gt; <span style="color: #00BB00;">mutate_p_value ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>      Package    Function.used                Time</span>
<span class="co">#&gt; 1 massdataset mutate_p_value() 2022-03-07 17:05:59</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="pathway-enrichment">
<strong>Pathway enrichment</strong><a class="anchor" aria-label="anchor" href="#pathway-enrichment"></a>
</h2>
<p>All the results will be placed in a folder named as <code>pathway_enrichment</code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span>path <span class="op">=</span> <span class="st">"pathway_enrichment"</span>, showWarnings <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">diff_metabolites</span> <span class="op">&lt;-</span>
  <span class="va">object_final</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu">activate_mass_dataset</span><span class="op">(</span>what <span class="op">=</span> <span class="st">"variable_info"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">p_value_adjust</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu">extract_variable_info</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">diff_metabolites</span><span class="op">)</span>
<span class="co">#&gt;     variable_id        mz        rt    na_freq  na_freq.1  na_freq.2 na_freq_2</span>
<span class="co">#&gt; 1    M86T95_POS  86.09716  94.57264 0.02564103 0.55454545 0.29090909        NA</span>
<span class="co">#&gt; 2 M95T100_1_POS  95.04975  99.77637 0.00000000 0.00000000 0.00000000        NA</span>
<span class="co">#&gt; 3  M103T100_POS 103.05477  99.90601 0.00000000 0.00000000 0.00000000        NA</span>
<span class="co">#&gt; 4   M104T51_POS 104.10746  51.27993 0.00000000 0.07272727 0.00000000        NA</span>
<span class="co">#&gt; 5   M113T81_POS 113.03501  80.73506 0.00000000 0.00000000 0.00000000        NA</span>
<span class="co">#&gt; 6  M113T187_POS 113.06018 186.56470 0.02564103 0.01818182 0.02727273        NA</span>
<span class="co">#&gt;   na_freq.1_2 na_freq.2_2       fc      p_value p_value_adjust</span>
<span class="co">#&gt; 1          NA          NA 1.714629 3.034938e-20   4.702229e-19</span>
<span class="co">#&gt; 2          NA          NA 1.439791 6.761432e-31   3.935153e-29</span>
<span class="co">#&gt; 3          NA          NA 1.345809 5.215317e-19   6.898442e-18</span>
<span class="co">#&gt; 4          NA          NA 1.751085 2.695171e-08   1.188325e-07</span>
<span class="co">#&gt; 5          NA          NA 1.500756 5.966911e-22   1.335670e-20</span>
<span class="co">#&gt; 6          NA          NA 1.951417 1.848299e-17   1.854673e-16</span>
<span class="co">#&gt;          Compound.name    CAS.ID   HMDB.ID KEGG.ID     Lab.ID     Adduct</span>
<span class="co">#&gt; 1           Piperidine  110-89-4      &lt;NA&gt;  C01746  MONA_2852     (M+H)+</span>
<span class="co">#&gt; 2               Phenol  108-95-2      &lt;NA&gt;  D01960 MONA_18506     (M+H)+</span>
<span class="co">#&gt; 3   Phenylacetaldehyde  122-78-1 HMDB06236  C00601    NO07389 (M+H-H2O)+</span>
<span class="co">#&gt; 4   5-Amino-1-pentanol 2508-29-4      &lt;NA&gt;    &lt;NA&gt;    NO07238     (M+H)+</span>
<span class="co">#&gt; 5               URACIL      &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt; MONA_18148     (M+H)+</span>
<span class="co">#&gt; 6 1,4-Cyclohexanedione      &lt;NA&gt;      &lt;NA&gt;    &lt;NA&gt; MONA_14519     (M+H)+</span>
<span class="co">#&gt;   mz.error mz.match.score RT.error RT.match.score                  CE        SS</span>
<span class="co">#&gt; 1 1.746869      0.9932417       NA             NA                  30 0.6143541</span>
<span class="co">#&gt; 2 1.416428      0.9955515       NA             NA                  10 0.6102452</span>
<span class="co">#&gt; 3 1.537004      0.9947640       NA             NA                  10 0.5748835</span>
<span class="co">#&gt; 4 1.169128      0.9969671       NA             NA                   5 0.5971697</span>
<span class="co">#&gt; 5 1.275544      0.9963909       NA             NA                  10 0.6889885</span>
<span class="co">#&gt; 6 1.051626      0.9975454       NA             NA HCD (NCE 20-30-40%) 0.5401414</span>
<span class="co">#&gt;   Total.score   Database Level</span>
<span class="co">#&gt; 1   0.7564369 MoNA_0.0.1     2</span>
<span class="co">#&gt; 2   0.7547351 MoNA_0.0.1     2</span>
<span class="co">#&gt; 3   0.7323387 NIST_0.0.1     2</span>
<span class="co">#&gt; 4   0.7470937 NIST_0.0.1     2</span>
<span class="co">#&gt; 5   0.8042644 MoNA_0.0.1     2</span>
<span class="co">#&gt; 6   0.7116679 MoNA_0.0.1     2</span></code></pre></div>
<div class="section level3">
<h3 id="load-kegg-human-pathway-database">Load <code>KEGG</code> human pathway database<a class="anchor" aria-label="anchor" href="#load-kegg-human-pathway-database"></a>
</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"kegg_hsa_pathway"</span>, package <span class="op">=</span> <span class="st">"metpath"</span><span class="op">)</span>
<span class="va">kegg_hsa_pathway</span>
<span class="co">#&gt; <span style="color: #00BB00;">---------Pathway source&amp;version---------</span></span>
<span class="co"><span style="color: #00BB00;">#&gt; KEGG &amp; 2021-12-13 </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; -----------Pathway information------------</span></span>
<span class="co"><span style="color: #00BB00;">#&gt; 345 pathways </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; 334 pathways have genes </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; 0 pathways have proteins </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; 281 pathways have compounds </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; Pathway class (top 10): Metabolism; Carbohydrate metabolism;Metabolism; Lipid metabolism </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">get_pathway_class</span><span class="op">(</span><span class="va">kegg_hsa_pathway</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 43 Ã— 2</span></span>
<span class="co">#&gt;    class                                                                       n</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                                                   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Cellular Processes; Cell growth and death                                   8</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Cellular Processes; Cell motility                                           1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> Cellular Processes; Cellular community - eukaryotes                         5</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Cellular Processes; Transport and catabolism                                7</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Environmental Information Processing; Membrane transport                    1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Environmental Information Processing; Signal transduction                  26</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Environmental Information Processing; Signaling molecules and interactâ€¦     5</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> Genetic Information Processing; Folding, sorting and degradation            7</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Genetic Information Processing; Replication and repair                      7</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Genetic Information Processing; Transcription                               3</span>
<span class="co">#&gt; <span style="color: #949494;"># â€¦ with 33 more rows</span></span></code></pre></div>
<p>Remove the disease pathways:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#get the class of pathways</span>
<span class="va">pathway_class</span> <span class="op">=</span> 
  <span class="fu">metpath</span><span class="fu">::</span><span class="fu">pathway_class</span><span class="op">(</span><span class="va">kegg_hsa_pathway</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pathway_class</span><span class="op">)</span>
<span class="co">#&gt; $hsa00010</span>
<span class="co">#&gt; [1] "Metabolism; Carbohydrate metabolism"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $hsa00020</span>
<span class="co">#&gt; [1] "Metabolism; Carbohydrate metabolism"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $hsa00030</span>
<span class="co">#&gt; [1] "Metabolism; Carbohydrate metabolism"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $hsa00040</span>
<span class="co">#&gt; [1] "Metabolism; Carbohydrate metabolism"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $hsa00051</span>
<span class="co">#&gt; [1] "Metabolism; Carbohydrate metabolism"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $hsa00052</span>
<span class="co">#&gt; [1] "Metabolism; Carbohydrate metabolism"</span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">remain_idx</span> <span class="op">=</span>
  <span class="va">pathway_class</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="st">"Disease"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu">`!`</span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">remain_idx</span>
<span class="co">#&gt;   [1]   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18</span>
<span class="co">#&gt;  [19]  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36</span>
<span class="co">#&gt;  [37]  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54</span>
<span class="co">#&gt;  [55]  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72</span>
<span class="co">#&gt;  [73]  73  74  75  76  77  78  79  80  81  82  83  84  85  90  91  92  93  94</span>
<span class="co">#&gt;  [91]  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112</span>
<span class="co">#&gt; [109] 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130</span>
<span class="co">#&gt; [127] 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148</span>
<span class="co">#&gt; [145] 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166</span>
<span class="co">#&gt; [163] 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184</span>
<span class="co">#&gt; [181] 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202</span>
<span class="co">#&gt; [199] 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220</span>
<span class="co">#&gt; [217] 221 222 223 224 225 226 227 228 229 230 236 240 241 242 243 244 245 246</span>
<span class="co">#&gt; [235] 247 248 249 250 251 252 253 254</span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pathway_database</span> <span class="op">=</span>
  <span class="va">kegg_hsa_pathway</span><span class="op">[</span><span class="va">remain_idx</span><span class="op">]</span>
<span class="va">pathway_database</span>
<span class="co">#&gt; <span style="color: #00BB00;">---------Pathway source&amp;version---------</span></span>
<span class="co"><span style="color: #00BB00;">#&gt; KEGG &amp; 2021-12-13 </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; -----------Pathway information------------</span></span>
<span class="co"><span style="color: #00BB00;">#&gt; 242 pathways </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; 235 pathways have genes </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; 0 pathways have proteins </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; 191 pathways have compounds </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; Pathway class (top 10): Metabolism; Carbohydrate metabolism;Metabolism; Lipid metabolism </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span></span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">kegg_id</span> <span class="op">&lt;-</span>
  <span class="va">diff_metabolites</span><span class="op">$</span><span class="va">KEGG.ID</span> 
<span class="va">kegg_id</span> <span class="op">&lt;-</span>
  <span class="va">kegg_id</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">kegg_id</span><span class="op">)</span><span class="op">]</span>
<span class="va">kegg_id</span>
<span class="co">#&gt;  [1] "C01746" "D01960" "C00601" "C00153" "C01108" "C00906" "C10438" "C00300"</span>
<span class="co">#&gt;  [9] "C00407" "C14790" "C08493" "C02237" "C01575" "C00073" "C05842" "C00637"</span>
<span class="co">#&gt; [17] "C00079" "D00022" "C07481" "C12305" "C17846" "D00029" "C00399" "C08362"</span>
<span class="co">#&gt; [25] "C14214" "C01595" "C00319" "C00410" "C10523" "C01780" "C00762" "C00735"</span>
<span class="co">#&gt; [33] "C17337" "C01921" "C04230" "C06539" "C00186" "C01546" "C00490" "C02226"</span>
<span class="co">#&gt; [41] "C00064" "C06104" "C02612" "C07599" "C05593" "0"      "C01601" "C02656"</span>
<span class="co">#&gt; [49] "C00366" "C07130" "C05635" "C16038" "C08322" "C16308" "C10911" "C05498"</span>
<span class="co">#&gt; [57] "C05472" "C04555"</span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">result</span> <span class="op">&lt;-</span> 
<span class="fu">enrich_kegg</span><span class="op">(</span>query_id <span class="op">=</span> <span class="va">kegg_id</span>, 
            query_type <span class="op">=</span> <span class="st">"compound"</span>, 
            id_type <span class="op">=</span> <span class="st">"KEGG"</span>,
            pathway_database <span class="op">=</span> <span class="va">pathway_database</span>, 
            p_cutoff <span class="op">=</span> <span class="fl">0.05</span>, 
            p_adjust_method <span class="op">=</span> <span class="st">"BH"</span>, 
            threads <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #00BB00;">191 pathways.</span></span></code></pre></div>
<p>Check the result:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">result</span>
<span class="co">#&gt; <span style="color: #00BB00;">---------Pathway database&amp;version---------</span></span>
<span class="co"><span style="color: #00BB00;">#&gt; KEGG &amp; 2021-12-13 </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; -----------Enrichment result------------</span></span>
<span class="co"><span style="color: #00BB00;">#&gt; 191 pathways are enriched </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; 11 pathways p-values &lt; 0.05 </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; Glycolysis / Gluconeogenesis;Citrate cycle (TCA cycle);Pentose phosphate pathway;Pentose and glucuronate interconversions;Fructose and mannose metabolism ... (only top 5 shows)</span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="plot-to-show-pathway-enrichment-result">
<strong>Plot to show pathway enrichment result</strong><a class="anchor" aria-label="anchor" href="#plot-to-show-pathway-enrichment-result"></a>
</h2>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">enrich_bar_plot</span><span class="op">(</span>object <span class="op">=</span> <span class="va">result</span>,
                x_axis <span class="op">=</span> <span class="st">"p_value"</span>,
                cutoff <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></code></pre></div>
<p><img src="pathway_enrichment_files/figure-html/unnamed-chunk-16-1.png" width="100%"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">enrich_scatter_plot</span><span class="op">(</span>object <span class="op">=</span> <span class="va">result</span>, y_axis <span class="op">=</span> <span class="st">"p_value"</span>, y_axis_cutoff <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></code></pre></div>
<p><img src="pathway_enrichment_files/figure-html/unnamed-chunk-17-1.png" width="100%"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">enrich_network</span><span class="op">(</span>
  object <span class="op">=</span> <span class="va">result</span>,
  point_size <span class="op">=</span> <span class="st">"p_value"</span>,
  p_cutoff <span class="op">=</span> <span class="fl">0.05</span>,
  only_significant_pathway <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span></code></pre></div>
<p><img src="pathway_enrichment_files/figure-html/unnamed-chunk-18-1.png" width="100%"></p>
</div>
<div class="section level2">
<h2 id="session-information">
<strong>Session information</strong><a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.2 (2021-11-01)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Big Sur 10.16</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">#&gt; [8] base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] forcats_0.5.1.9000   stringr_1.4.0        purrr_0.3.4         </span>
<span class="co">#&gt;  [4] readr_2.1.2          tidyr_1.2.0          tibble_3.1.6        </span>
<span class="co">#&gt;  [7] tidyverse_1.3.1      tinytools_0.9.1      dplyr_1.0.8         </span>
<span class="co">#&gt; [10] metid_1.2.4          metpath_0.99.2       massstat_0.99.12    </span>
<span class="co">#&gt; [13] ggfortify_0.4.14     massqc_0.99.7        masscleaner_0.99.7  </span>
<span class="co">#&gt; [16] xcms_3.16.1          MSnbase_2.20.4       ProtGenerics_1.26.0 </span>
<span class="co">#&gt; [19] S4Vectors_0.32.3     mzR_2.28.0           Rcpp_1.0.8          </span>
<span class="co">#&gt; [22] Biobase_2.54.0       BiocGenerics_0.40.0  BiocParallel_1.28.3 </span>
<span class="co">#&gt; [25] massprocesser_0.99.3 ggplot2_3.3.5        magrittr_2.0.2      </span>
<span class="co">#&gt; [28] masstools_0.99.4     massdataset_0.99.14  tidymass_0.99.5     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] ragg_1.2.1                  missForest_1.4             </span>
<span class="co">#&gt;   [3] knitr_1.37                  DelayedArray_0.20.0        </span>
<span class="co">#&gt;   [5] data.table_1.14.2           rpart_4.1.16               </span>
<span class="co">#&gt;   [7] KEGGREST_1.34.0             RCurl_1.98-1.5             </span>
<span class="co">#&gt;   [9] doParallel_1.0.17           generics_0.1.2             </span>
<span class="co">#&gt;  [11] snow_0.4-4                  leaflet_2.1.0              </span>
<span class="co">#&gt;  [13] preprocessCore_1.56.0       mixOmics_6.18.1            </span>
<span class="co">#&gt;  [15] RANN_2.6.1                  proxy_0.4-26               </span>
<span class="co">#&gt;  [17] future_1.23.0               tzdb_0.2.0                 </span>
<span class="co">#&gt;  [19] xml2_1.3.3                  lubridate_1.8.0            </span>
<span class="co">#&gt;  [21] ggsci_2.9                   SummarizedExperiment_1.24.0</span>
<span class="co">#&gt;  [23] assertthat_0.2.1            viridis_0.6.2              </span>
<span class="co">#&gt;  [25] xfun_0.29                   hms_1.1.1                  </span>
<span class="co">#&gt;  [27] jquerylib_0.1.4             evaluate_0.15              </span>
<span class="co">#&gt;  [29] DEoptimR_1.0-10             fansi_1.0.2                </span>
<span class="co">#&gt;  [31] dbplyr_2.1.1                readxl_1.3.1               </span>
<span class="co">#&gt;  [33] igraph_1.2.11               DBI_1.1.2                  </span>
<span class="co">#&gt;  [35] htmlwidgets_1.5.4           MsFeatures_1.3.0           </span>
<span class="co">#&gt;  [37] rARPACK_0.11-0              ellipsis_0.3.2             </span>
<span class="co">#&gt;  [39] RSpectra_0.16-0             crosstalk_1.2.0            </span>
<span class="co">#&gt;  [41] backports_1.4.1             ggcorrplot_0.1.3           </span>
<span class="co">#&gt;  [43] MatrixGenerics_1.6.0        vctrs_0.3.8                </span>
<span class="co">#&gt;  [45] remotes_2.4.2               cachem_1.0.6               </span>
<span class="co">#&gt;  [47] withr_2.4.3                 ggforce_0.3.3              </span>
<span class="co">#&gt;  [49] itertools_0.1-3             robustbase_0.93-9          </span>
<span class="co">#&gt;  [51] checkmate_2.0.0             cluster_2.1.2              </span>
<span class="co">#&gt;  [53] lazyeval_0.2.2              crayon_1.5.0               </span>
<span class="co">#&gt;  [55] ellipse_0.4.2               labeling_0.4.2             </span>
<span class="co">#&gt;  [57] pkgconfig_2.0.3             tweenr_1.0.2               </span>
<span class="co">#&gt;  [59] GenomeInfoDb_1.30.0         nnet_7.3-17                </span>
<span class="co">#&gt;  [61] rlang_1.0.1                 globals_0.14.0             </span>
<span class="co">#&gt;  [63] lifecycle_1.0.1             affyio_1.64.0              </span>
<span class="co">#&gt;  [65] extrafontdb_1.0             fastDummies_1.6.3          </span>
<span class="co">#&gt;  [67] MassSpecWavelet_1.60.0      modelr_0.1.8               </span>
<span class="co">#&gt;  [69] cellranger_1.1.0            randomForest_4.7-1         </span>
<span class="co">#&gt;  [71] rprojroot_2.0.2             polyclip_1.10-0            </span>
<span class="co">#&gt;  [73] matrixStats_0.61.0          Matrix_1.4-0               </span>
<span class="co">#&gt;  [75] reprex_2.0.1                base64enc_0.1-3            </span>
<span class="co">#&gt;  [77] GlobalOptions_0.1.2         png_0.1-7                  </span>
<span class="co">#&gt;  [79] viridisLite_0.4.0           rjson_0.2.21               </span>
<span class="co">#&gt;  [81] clisymbols_1.2.0            bitops_1.0-7               </span>
<span class="co">#&gt;  [83] pander_0.6.4                Biostrings_2.62.0          </span>
<span class="co">#&gt;  [85] shape_1.4.6                 parallelly_1.30.0          </span>
<span class="co">#&gt;  [87] robust_0.7-0                jpeg_0.1-9                 </span>
<span class="co">#&gt;  [89] gridGraphics_0.5-1          scales_1.1.1               </span>
<span class="co">#&gt;  [91] memoise_2.0.1               plyr_1.8.6                 </span>
<span class="co">#&gt;  [93] zlibbioc_1.40.0             compiler_4.1.2             </span>
<span class="co">#&gt;  [95] RColorBrewer_1.1-2          pcaMethods_1.86.0          </span>
<span class="co">#&gt;  [97] clue_0.3-60                 rrcov_1.6-2                </span>
<span class="co">#&gt;  [99] cli_3.2.0                   affy_1.72.0                </span>
<span class="co">#&gt; [101] XVector_0.34.0              listenv_0.8.0              </span>
<span class="co">#&gt; [103] patchwork_1.1.1             pbapply_1.5-0              </span>
<span class="co">#&gt; [105] htmlTable_2.4.0             Formula_1.2-4              </span>
<span class="co">#&gt; [107] MASS_7.3-55                 tidyselect_1.1.1           </span>
<span class="co">#&gt; [109] vsn_3.62.0                  stringi_1.7.6              </span>
<span class="co">#&gt; [111] textshaping_0.3.6           highr_0.9                  </span>
<span class="co">#&gt; [113] yaml_2.3.4                  latticeExtra_0.6-29        </span>
<span class="co">#&gt; [115] MALDIquant_1.21             ggrepel_0.9.1              </span>
<span class="co">#&gt; [117] grid_4.1.2                  sass_0.4.0                 </span>
<span class="co">#&gt; [119] tools_4.1.2                 parallel_4.1.2             </span>
<span class="co">#&gt; [121] circlize_0.4.14             rstudioapi_0.13            </span>
<span class="co">#&gt; [123] MsCoreUtils_1.6.0           foreach_1.5.2              </span>
<span class="co">#&gt; [125] foreign_0.8-82              gridExtra_2.3              </span>
<span class="co">#&gt; [127] farver_2.1.0                mzID_1.32.0                </span>
<span class="co">#&gt; [129] ggraph_2.0.5                rvcheck_0.2.1              </span>
<span class="co">#&gt; [131] digest_0.6.29               BiocManager_1.30.16        </span>
<span class="co">#&gt; [133] GenomicRanges_1.46.1        broom_0.7.12               </span>
<span class="co">#&gt; [135] ncdf4_1.19                  httr_1.4.2                 </span>
<span class="co">#&gt; [137] ComplexHeatmap_2.10.0       colorspace_2.0-2           </span>
<span class="co">#&gt; [139] rvest_1.0.2                 XML_3.99-0.8               </span>
<span class="co">#&gt; [141] fs_1.5.2                    IRanges_2.28.0             </span>
<span class="co">#&gt; [143] splines_4.1.2               yulab.utils_0.0.4          </span>
<span class="co">#&gt; [145] graphlayouts_0.8.0          pkgdown_2.0.2              </span>
<span class="co">#&gt; [147] ggplotify_0.1.0             plotly_4.10.0              </span>
<span class="co">#&gt; [149] systemfonts_1.0.3           fit.models_0.64            </span>
<span class="co">#&gt; [151] jsonlite_1.7.3              tidygraph_1.2.0            </span>
<span class="co">#&gt; [153] corpcor_1.6.10              R6_2.5.1                   </span>
<span class="co">#&gt; [155] Hmisc_4.6-0                 pillar_1.7.0               </span>
<span class="co">#&gt; [157] htmltools_0.5.2             glue_1.6.1                 </span>
<span class="co">#&gt; [159] fastmap_1.1.0               class_7.3-20               </span>
<span class="co">#&gt; [161] codetools_0.2-18            pcaPP_1.9-74               </span>
<span class="co">#&gt; [163] mvtnorm_1.1-3               furrr_0.2.3                </span>
<span class="co">#&gt; [165] utf8_1.2.2                  lattice_0.20-45            </span>
<span class="co">#&gt; [167] bslib_0.3.1                 zip_2.2.0                  </span>
<span class="co">#&gt; [169] openxlsx_4.2.5              Rttf2pt1_1.3.9             </span>
<span class="co">#&gt; [171] survival_3.2-13             limma_3.50.0               </span>
<span class="co">#&gt; [173] rmarkdown_2.11              desc_1.4.0                 </span>
<span class="co">#&gt; [175] munsell_0.5.0               e1071_1.7-9                </span>
<span class="co">#&gt; [177] GetoptLong_1.0.5            GenomeInfoDbData_1.2.7     </span>
<span class="co">#&gt; [179] iterators_1.0.14            impute_1.68.0              </span>
<span class="co">#&gt; [181] haven_2.4.3                 reshape2_1.4.4             </span>
<span class="co">#&gt; [183] gtable_0.3.0                extrafont_0.17</span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>tidymass Developed by Xiaotao Shen. <br> from <i class="fa fa-university"></i> Stanford University <br><i class="fa fa-clinic-medical"></i>School of Medicine <br><i class="fa fa-dna"></i> Genetics Department <br><i class="fa fa-laptop-code"></i> <a href="https://www.shenxt.info/files/wechat_QR.jpg" class="external-link">More softare</a></p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p><b>Contact Us:</b> <br><i class="fa fa-envelope"></i> <a href="mailto:shenxt1990@outlook.com" class="email">shenxt1990@outlook.com</a> <br><i class="fa fa-weixin"></i> <a href="https://www.shenxt.info/files/wechat_QR.jpg" class="external-link">shenxt1990</a> <br><i class="fa fa-home"></i> <a href="https://www.shenxt.info" class="external-link">shenxt.info</a> <br><i class="fa fa-twitter"></i> <a href="https://twitter.com/xiaotaoshen1990" class="external-link">Twitter</a> <br><i class="fa fa-map-marker-alt"></i> <a href="https://goo.gl/maps/WLps4yacxfRfyAxTA" class="external-link">BMI building</a></p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
